users/helper.py

```diff
@@ -0,0 +1,152 @@
+        'page_impressions_by_age_gender_unique',
+        'page_impressions_viral_frequency_distribution'
+    ]
+
+    post_insights_metrices = [
+        'page_posts_impressions',
+        'page_posts_impressions_unique',
+        'page_posts_impressions_paid',
+        'page_posts_impressions_paid_unique',
+        'page_posts_impressions_organic',
+        'page_posts_impressions_organic_unique',
+        'page_posts_served_impressions_organic_unique',
+        'page_posts_impressions_viral',
+        'page_posts_impressions_viral_unique',
+        'page_posts_impressions_nonviral',
+        'page_posts_impressions_nonviral_unique',
+        'post_engaged_users',
+        'post_negative_feedback',
+        'post_negative_feedback_unique',
+        'post_negative_feedback_by_type',
+        'post_negative_feedback_by_type_unique',
+        'post_engaged_fan',
+        'post_clicks',
+        'post_clicks_unique',
+        'post_clicks_by_type',
+        'post_clicks_by_type_unique',
+        'post_impressions',
+        'post_impressions_unique',
+        'post_impressions_paid',
+        'post_impressions_paid_unique',
+        'post_impressions_fan',
+        'post_impressions_fan_unique',
+        'post_impressions_organic',
+        'post_impressions_organic_unique',
+        'post_impressions_viral',
+        'post_impressions_viral_unique',
+        'post_impressions_nonviral',
+        'post_impressions_nonviral_unique',
+        'post_impressions_by_story_type',
+        'post_impressions_by_story_type_unique',
+        'post_impressions_by_paid_non_paid',
+        'post_reactions_like_total',
+        'post_reactions_love_total',
+        'post_reactions_wow_total',
+        'post_reactions_haha_total',
+        'post_reactions_sorry_total',
+        'post_reactions_anger_total',
+        'post_reactions_by_type_total'
+    ]
+
+    def get_user_access_token_and_id(user):
+        try:
+            social = user.social_auth.get(provider='facebook')
+            access_token = social.extra_data['access_token']
+            user_id = social.extra_data.get('id', None)
+            return access_token, user_id
+        except UserSocialAuth.DoesNotExist:
+            return None
+
+    def get_facebook_page_info(user):
+        access_token, user_id = FacebookClient.get_user_access_token_and_id(user)
+
+        if access_token and user_id:
+            url = f"https://graph.facebook.com/{user_id}/accounts"
+            params = {'access_token': access_token}
+            response = requests.get(url, params=params)
+
+            if response.status_code == 200:
+                data = response.json()
+                if 'data' in data and len(data['data']) > 0:
+                    page_info = data['data'][0]
+                    return page_info.get('access_token'), page_info.get('id')
+                else:
+                    return None
+            else:
+                return None
+        else:
+            return None
```

Comment:
these else statements aren't doing much and wouldn't make difference if you remove them, also this much nesting should be avoided by early exit